pkg/windows/service/service.go

```diff
@@ -0,0 +1,235 @@
+		ActionsCount uint32
+		Actions      uintptr
+	}
+
+	type scAction struct {
+		Type  uint32
+		Delay uint32
+	}
+	t := []scAction{
+		{Type: scActionRestart, Delay: uint32(60 * time.Second / time.Millisecond)},
+		{Type: scActionRestart, Delay: uint32(600 * time.Second / time.Millisecond)},
+		{Type: scActionNone},
+	}
+	lpInfo := serviceFailureActions{ResetPeriod: uint32(24 * time.Hour / time.Second), ActionsCount: uint32(3), Actions: uintptr(unsafe.Pointer(&t[0]))}
+	err = windows.ChangeServiceConfig2(s.Handle, serviceConfigFailureActions, (*byte)(unsafe.Pointer(&lpInfo)))
+	if err == nil {
+		glog.Infof("Service successfully registered!")
+	}
+	return err
+}
+
+func unregisterService() error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(*flServiceName)
+	if err != nil {
+		return err
+	}
+	defer s.Close()
+
+	err = s.Delete()
+	if err != nil {
+		return err
+	}
+	glog.Infof("Service successfully unregistered!")
+	return nil
+}
+
+// InitService is the entry point for running the daemon as a Windows
+// service. It returns an indication to stop (if registering/un-registering);
+// an indication of whether it is running as a service; and an error.
+func InitService() (bool, bool, error) {
+	if *flUnregisterService {
+		if *flRegisterService {
+			return true, false, fmt.Errorf("--register-service and --unregister-service cannot be used together")
+		}
+		return true, false, unregisterService()
+	}
+
+	if *flRegisterService {
+		return true, false, registerService()
+	}
+
+	if !*flRunService {
+		return false, false, nil
+	}
+
+	h := &handler{
+		tosvc:   make(chan bool),
+		fromsvc: make(chan error),
+	}
+
+	service = h
+	var err error
+	go func() {
+		err = svc.Run(*flServiceName, h)
+		h.fromsvc <- err
+	}()
+
+	// Wait for the first signal from the service handler.
+	err = <-h.fromsvc
+	if err != nil {
+		return false, false, err
+	}
+	return false, true, nil
+}
+
+func (h *handler) Execute(_ []string, r <-chan svc.ChangeRequest, s chan<- svc.Status) (bool, uint32) {
+	s <- svc.Status{State: svc.StartPending, Accepts: 0}
+	// Unblock initService()
+	h.fromsvc <- nil
+
+	s <- svc.Status{State: svc.Running, Accepts: svc.AcceptStop | svc.AcceptShutdown | svc.Accepted(windows.SERVICE_ACCEPT_PARAMCHANGE)}
+	glog.Infof("Service running")
+Loop:
+	for {
+		select {
+		case <-h.tosvc:
+			break Loop
+		case c := <-r:
+			switch c.Cmd {
+			case svc.Cmd(windows.SERVICE_CONTROL_PARAMCHANGE):
+				s <- c.CurrentStatus
+			case svc.Interrogate:
+				s <- c.CurrentStatus
+			case svc.Stop, svc.Shutdown:
+				s <- svc.Status{State: svc.Stopped}
+				// TODO: Stop the kubelet gracefully instead of killing the process
+				os.Exit(0)
+			}
+		}
+	}
+
+	return false, 0
+}
+
+// InstallServiceFlags adds additional flags for the Windows binary to
+// enable the registration as a service
+func InstallServiceFlags(flags *pflag.FlagSet, serviceName string, serviceDisplayName string) {
```

Comment:
if any of these flags are considered "alpha," please include `<Warning: Alpha feature>` at the beginning of the usage text

Reply:
The flags are not considered "alpha". They are doing a very simple job of registering/removing the kubelet/kube-proxy as Windows services and nothing more.